src/panels/config/integrations/ha-config-integrations.ts

```diff
@@ -677,49 +759,92 @@ class HaConfigIntegrations extends SubscribeMixin(LitElement) {
     if (!domain) {
       return;
     }
+
     const handlers = await getConfigFlowHandlers(this.hass, "integration");
 
-    if (!handlers.includes(domain)) {
-      if (HELPER_DOMAINS.includes(domain)) {
-        navigate(`/config/helpers/add?domain=${domain}`, {
-          replace: true,
-        });
+    // Integration exists, so we can just create a flow
+    if (handlers.includes(domain)) {
+      const localize = await localizePromise;
+      if (
+        !(await showConfirmationDialog(this, {
+          title: localize("ui.panel.config.integrations.confirm_new", {
+            integration: domainToName(localize, domain),
+          }),
+        }))
+      ) {
         return;
       }
-      const helpers = await getConfigFlowHandlers(this.hass, "helper");
-      if (helpers.includes(domain)) {
-        navigate(`/config/helpers/add?domain=${domain}`, {
-          replace: true,
-        });
-        return;
+      showConfigFlowDialog(this, {
+        dialogClosedCallback: () => {
+          this._handleFlowUpdated();
+        },
+        startFlowHandler: domain,
+        manifest: this._manifests[domain],
+        showAdvanced: this.hass.userData?.showAdvanced,
+      });
+    }
+
+    const supportedBrands = await getSupportedBrands(this.hass);
+    const supportedBrandsIntegrations = {};
+
+    for (const [d, domainBrands] of Object.entries(supportedBrands)) {
+      for (const [slug, name] of Object.entries(domainBrands)) {
+        supportedBrandsIntegrations[slug] = {
+          name,
+          supported_flows: [d],
+        };
       }
```

Comment:
Didn't we create a helper for this? If not we probably should

Reply:
Its not the same as the other need. But I did make one for this